xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -187,7 +188,7 @@ public boolean toDocument(Document doc, org.xwiki.rest.model.jaxb.Page restPage)
 
             // Remove objects
             List<com.xpn.xwiki.api.Object> toRemove = new ArrayList<>();
-            for (Vector<com.xpn.xwiki.api.Object> objects : doc.getxWikiObjects().values()) {
+            for (ArrayList<com.xpn.xwiki.api.Object> objects : doc.getxWikiObjects().values()) {
```

Comment:
It's not that simply and it shows that you did not built to validate your change.